File: mobiletrm-backend/src/main/java/world/inetumrealdolmen/dto/TaskDto.java
package world.inetumrealdolmen.dto;

import world.inetumrealdolmen.domain.Task;

/**
 * A collection of Data Transfer Objects (DTO) for {@link Task} entities to be sent to clients.
 * Use the static subclasses' constructors to create DTO's.
 */
public class TaskDto {
  /**
   * Represents an index view of a task.
   */
  public static class Index {

    /**
     * The ID of the Task.
     */
    public long id;
    /**
     * The name of the task.
     */
    public String name;

    /**
     * Default constructor for Jackson serializing.
     * Protected to prevent instantiation without parameters.
     */
    public Index() {
    }

    /**
     * Constructor to convert a Task object into a DTO with basic information.
     *
     * @param task The Task object to convert.
     */
    public Index(Task task) {
      id = task.id;
      name = task.name;
    }
  }
}

File: mobiletrm-android/app/src/main/java/world/inetumrealdolmen/mobiletrm/data/dto/TaskDTO.kt
package world.inetumrealdolmen.mobiletrm.data.dto

import kotlinx.serialization.Serializable
import world.inetumrealdolmen.mobiletrm.data.model.Task

/**
 * Index information of a task.
 *
 * @param id The database ID of the task.
 * @param name The name of the task.
 */
@Serializable
data class TaskDTO(
    var id: Long,
    var name: String,
)

/**
 * Converts the response data to a list of [Task]
 */
fun List<TaskDTO>.asDomainObjects(): List<Task> =
    map {
        it.asDomainObject()
    }

/**
 * Converts the response data to the [Task] model.
 */
fun TaskDTO.asDomainObject(): Task =
    Task(
        taskId = id,
        taskName = name,
    )
